### DISCUSSION TITLE
Need advice: Async atom fetch pattern with selectors

### DISCUSSION THREAD

**Discussion by simon3495** [pmndrs/jotai#2674] [Q&A]
I want to do the simplest of things:

1. Fetch some data and store it in an async atom. (Expect a json back from the api)
2. Now I want selectors (focusAtom?) that subscribes to only a part of the json data and that are able to update that part as well. 

What does this pattern look like in jotai? (I would prefer to render fallbacks using suspense)

There is no example of this in the docs: https://jotai.org/docs/guides/async

**Comment by dai-shi** [comment]
Do you mean something like this?

```ts
const dataAtom = atom((get) => Promise.resolve({ text: 'foo' }));
const textAtom = atom(async (get) => (await get(dataAtom)).text);
```

**Reply by simon3495** [reply]
But with that you cannot update the textAtom, you can only read it:
![image](https://github.com/user-attachments/assets/53c06061-3ec7-4e3e-bf61-a2acdad8908a)

So exactly what you wrote but, additionally, it should also be possible to update to `textAtom` as I was trying to in the image.

**Reply by dai-shi** [reply]
You can use `atomWithDefault` util and a writeable derived atom.
